__tests__/nc_news_test.test.js

```diff
@@ -89,3 +89,73 @@ describe('GET /api/topics', () => {
             })
         });
     })
+    describe('GET /api/articles', () => {
+        test('Responds with status code 200 and an articles array, with article objects and correct length.' , () => {
+            return request(app)
+            .get("/api/articles")
+            .expect(200)
+            .then((response) => {
+                const articles = response.body.articles;
+                expect(Array.isArray(articles)).toBe(true);
+                articles.forEach((article) => {
+                    expect(typeof article).toBe("object")
+                })
+                expect(articles.length).toBe(13);
+            })
+        });
+        test('Returned article objects posses an additional property of \'comment_count\'', () => {
+            return request(app)
+            .get("/api/articles")
+            .expect(200)
+            .then((response) => {
+                const articles = response.body.articles;
+                articles.forEach((article) => {
+                    expect(article).toHaveProperty("comment_count");
+                }) 
+            })    
+        })
+        test('Property \'comment_count\', now has a value of the number of comments who share that article ID', () => {
+            return request(app)
+            .get("/api/articles")
+            .expect(200)
+            .then((response) => {
+                const articles = response.body.articles;
+                articles.forEach((article) => {
+                    if(article.article_id === 1) {
+                        expect(article.comment_count).toBe("11")  
+                    }
+                    if(article.article_id === 3) {
+                        expect(article.comment_count).toBe("2")  
+                    }
+                    if(article.article_id === 5) {
+                        expect(article.comment_count).toBe("2")  
+                    }
+                    if(article.article_id === 6) {
+                        expect(article.comment_count).toBe("1")  
+                    }
+                    if(article.article_id === 9) {
+                        expect(article.comment_count).toBe("2")  
+                    }
+                })   
+           })    
+        })
+        test('Test that property \'body\' is not present inside the article objects', () => {
+            return request(app)
+            .get("/api/articles")
+            .expect(200)
+            .then((response)=>{
+                const articles = response.body.articles;
+                articles.forEach((article) => {
+                    expect(article).not.toHaveProperty("body");
+                })
+            })
+        })
+        test('Responds with 400 status code and error message, if provided a malformed path', () =>{
```

Comment:
This would be a 404 not a 400. If you're already testing for a bad endpoint then you don't need to test it again.

Reply:
I'm not sure I understand this one, wouldn't it be 400, since the path provided wasn't valid, or malformed? I thought 404 was more suited when the resource the client is searching for does not exist, like an id that isn't present in the database.